src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1908,6 +1909,9 @@ private TypeParameterConstraintSyntax ParseTypeParameterConstraint()
                     questionToken = this.TryEatToken(SyntaxKind.QuestionToken);
 
                     return _syntaxFactory.ClassOrStructConstraint(SyntaxKind.ClassConstraint, classToken, questionToken);
+                case SyntaxKind.DefaultKeyword:
+                    var defaultToken = this.EatToken();
+                    return _syntaxFactory.DefaultConstraint(defaultToken);
```

Comment:
>_syntaxFactory.DefaultConstraint(defaultToken) [](start = 27, length = 46)

Should we check language version here?

Reply:
The version check is in `BindTypeParameterConstraints()`.

---
In reply to: [456659327](https://github.com/dotnet/roslyn/pull/45993#discussion_r456659327) [](ancestors = 456659327)

Reply:
> The version check is in BindTypeParameterConstraints().

I realized that. The question is why? Usually, for simple cases like that we perform language version checks in the parser.

---
In reply to: [456693574](https://github.com/dotnet/roslyn/pull/45993#discussion_r456693574) [](ancestors = 456693574,456659327)